web/packages/teleterm/src/ui/AppInitializer.tsx

```diff
@@ -23,13 +26,26 @@ export const AppInitializer: FC = props => {
     initializeApp();
   }, [initializeApp]);
 
-  if (!isUiVisible) {
-    return <Centered>Loading</Centered>;
-  }
-
-  return <>{props.children}</>;
+  return (
+    <>
+      <LayoutManager />
+      {!isUiReady && (
+        <Centered>
+          <Indicator delay="short" />
+        </Centered>
+      )}
+      <ModalsHost />
+    </>
+  );
 };
 
 const Centered = styled.div`
-  margin: auto;
+  display: flex;
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  justify-content: center;
+  align-items: center;
+  z-index: 2;
```

Comment:
I know that this is to make sure that the indicator of `ConnectionsIcon` isn't visible through the overlay, could you mention that here?

Long term we probably should create an object with z-indexes instead of arbitrarily using them in CSS. So instead of just typing `2` here we'd do `zIndex[2]` or `zIndex['aboveContent']`. From my experience, using z-index without some kind of a system on top of it leads to people using a multitude of values for it within the project.

Reply:
Nice pattern with the object, I haven't seen that.

I added the comment.